File: ProvaTecnicaTarget/faturamento_q04.py
import json

# Nome do arquivo JSON
arquivo_json = 'faturamento_q04.json'

# Inicializa os dados de faturamento
faturamento_inicial = {
    "SP": 67836.43,
    "RJ": 36678.66,
    "MG": 29229.88,
    "ES": 27165.48,
    "Outros": 19849.53
}

# Função para salvar dados no arquivo JSON
def salvar_dados(dados):
    with open(arquivo_json, 'w') as arquivo:
        json.dump(dados, arquivo, indent=4)

# Função para carregar dados do arquivo JSON
def carregar_dados():
    try:
        with open(arquivo_json, 'r') as arquivo:
            return json.load(arquivo)
    except FileNotFoundError:
        salvar_dados(faturamento_inicial)
        return faturamento_inicial

# Função para normalizar o nome do estado
def normalizar_estado(estado):
    return estado.strip().upper()

# Função para listar o faturamento
def listar_faturamento(dados):
    print("\nFaturamento por Estado:")
    for estado, valor in dados.items():
        print(f"{estado}: R${valor:.2f}")
    print()

# Função para calcular o percentual de faturamento de um estado
def calcular_percentual(dados, estado):
    estado = normalizar_estado(estado)
    total = sum(dados.values())
    if estado in dados:
        percentual = (dados[estado] / total) * 100
        print(f"\nPercentual de {estado}: {percentual:.2f}%\n")
    else:
        print("\nEstado não encontrado!\n")

# Função para corrigir o faturamento de um estado
def corrigir_faturamento(dados, estado, novo_valor):
    estado = normalizar_estado(estado)
    if estado in dados:
        dados[estado] = novo_valor
        salvar_dados(dados)
        print("\nFaturamento atualizado com sucesso!\n")
    else:
        print("\nEstado não encontrado!\n")

# Função principal do menu
def menu():
    dados = carregar_dados()

    while True:
        print("Menu de Opções:")
        print("1. Listar Faturamento")
        print("2. Calcular Faturamento")
        print("3. Corrigir Faturamento")
        print("4. Sair")

        opcao = input("Escolha uma opção: ")

        if opcao == '1':
            listar_faturamento(dados)
        elif opcao == '2':
            estado = input("Digite o estado (SP, RJ, MG, ES, Outros): ")
            calcular_percentual(dados, estado)
        elif opcao == '3':
            estado = input("Digite o estado (SP, RJ, MG, ES, Outros): ")
            try:
                novo_valor = float(input("Digite o novo valor de faturamento: "))
                corrigir_faturamento(dados, estado, novo_valor)
            except ValueError:
                print("\nValor inválido!\n")
        elif opcao == '4':
            print("Saindo do programa.")
            break
        else:
            print("\nOpção inválida!\n")

# Executa o menu
menu()

File: ProvaTecnicaTarget/faturamento_q04.java

import java.io.*;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class faturamento_q04 {
    private static final String FILE_NAME = "faturamento_q04.txt";

    public static void main(String[] args) {
        Map<String, Double> faturamento = carregarDados();

        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("Menu de Opções:");
            System.out.println("1. Listar Faturamento");
            System.out.println("2. Calcular Faturamento");
            System.out.println("3. Corrigir Faturamento");
            System.out.println("4. Sair");

            System.out.print("Escolha uma opção: ");
            String opcao = scanner.nextLine();

            switch (opcao) {
                case "1":
                    listarFaturamento(faturamento);
                    break;
                case "2":
                    System.out.print("Digite o estado (SP, RJ, MG, ES, Outros): ");
                    String estado = normalizarEstado(scanner.nextLine());
                    calcularPercentual(faturamento, estado);
                    break;
                case "3":
                    System.out.print("Digite o estado (SP, RJ, MG, ES, Outros): ");
                    String estadoParaCorrigir = normalizarEstado(scanner.nextLine());
                    System.out.print("Digite o novo valor de faturamento: ");
                    double novoValor = scanner.nextDouble();
                    scanner.nextLine();  // Clear buffer
                    corrigirFaturamento(faturamento, estadoParaCorrigir, novoValor);
                    break;
                case "4":
                    System.out.println("Saindo do programa.");
                    salvarDados(faturamento);
                    scanner.close();
                    return;
                default:
                    System.out.println("Opção inválida!");
            }
        }
    }

    // Função para normalizar o nome do estado
    private static String normalizarEstado(String estado) {
        return estado.trim().toUpperCase();
    }

    // Função para listar o faturamento
    private static void listarFaturamento(Map<String, Double> dados) {
        System.out.println("\nFaturamento por Estado:");
        for (Map.Entry<String, Double> entry : dados.entrySet()) {
            System.out.printf("%s: R$%.2f%n", entry.getKey(), entry.getValue());
        }
        System.out.println();
    }

    // Função para calcular o percentual de faturamento de um estado
    private static void calcularPercentual(Map<String, Double> dados, String estado) {
        double total = dados.values().stream().mapToDouble(Double::doubleValue).sum();
        if (dados.containsKey(estado)) {
            double percentual = (dados.get(estado) / total) * 100;
            System.out.printf("\nPercentual de %s: %.2f%%%n%n", estado, percentual);
        } else {
            System.out.println("\nEstado não encontrado!\n");
        }
    }

    // Função para corrigir o faturamento de um estado
    private static void corrigirFaturamento(Map<String, Double> dados, String estado, double novoValor) {
        if (dados.containsKey(estado)) {
            dados.put(estado, novoValor);
            System.out.println("\nFaturamento atualizado com sucesso!\n");
        } else {
            System.out.println("\nEstado não encontrado!\n");
        }
    }

    // Função para salvar dados no arquivo TXT
    private static void salvarDados(Map<String, Double> dados) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(FILE_NAME))) {
            for (Map.Entry<String, Double> entry : dados.entrySet()) {
                writer.printf("%s:%.2f%n", entry.getKey(), entry.getValue());
            }
        } catch (IOException e) {
            System.out.println("Erro ao salvar dados no arquivo: " + e.getMessage());
        }
    }

    // Função para carregar dados do arquivo TXT
    private static Map<String, Double> carregarDados() {
        Map<String, Double> dados = new HashMap<>();
        File file = new File(FILE_NAME);
        if (file.exists()) {
            try (Scanner scanner = new Scanner(file)) {
                while (scanner.hasNextLine()) {
                    String[] linha = scanner.nextLine().split(":");
                    dados.put(linha[0], Double.parseDouble(linha[1]));
                }
            } catch (FileNotFoundException e) {
                System.out.println("Erro ao abrir o arquivo: " + e.getMessage());
            }
        } else {
            dados.put("SP", 67836.43);
            dados.put("RJ", 36678.66);
            dados.put("MG", 29229.88);
            dados.put("ES", 27165.48);
            dados.put("Outros", 19849.53);
        }
        return dados;
    }
}

